### ISSUE TITLE
glide adds current package to glide.lock

### ISSUE THREAD

**Issue by terinjokes** [Masterminds/glide#322]
I'm running into an issue where the current package is getting added to glide.lock.

> Names changed to protect the innocent.

```
$ glide --version
glide version 0.9.1
```

```
GOPATH=/Users/terin/dev/foobar
PWD=/Users/terin/dev/foobar/src/github.com/terinjokes/foobar
```

The code at `$PWD/cmd/foobar/main.go` imports the package `github.com/terinjokes/foobar`.

```
$ glide name
github.com/terinjokes/foobar
```

I would expect glide to ignore the "github.com/terinjokes/foobar" import (as it's the current package), but it dutifully fetches master and adds the hash to glide.lock.

At this time I do not know how to reproduce this behavior in another project, but I will continue trying.

**Comment by mattfarina** [comment]
Can you share your `glide.yaml` file?

**Comment by technosophos** [comment]
Also maybe the output of `glide --debug up`?

**Comment by terinjokes** [comment]
OK. I think I've tracked down what happened.

Originally my package was "github.com/terinjokes/foo-bar", and was located as such underneath the `$GOPATH`. I developed the library first, later adding the CLI tool. As I was developing the CLI tool, the package was renamed to "github.com/terinjokes/foobar" and my CLI tool imported that path. I realized the "package" in `glide.yaml` was wrong, as well as my checkout under `$GOPATH`, so I corrected both. However, by this point the checkout had been created in `vendor/` and the item added to the lockfile.

Removing it from the lockfile would only result it in being readded the next time `glide` ran. I suppose since it was being imported, `glide` would attempt resolution, finding it under `vendor/` and thus adding it to the lockfile. Deleting it from both the lockfile and the vendor directory results it in remaining gone.
